core/htmldataprocessor.js

```diff
@@ -58,6 +58,11 @@
 
 			// The source data is already HTML, but we need to clean
 			// it up and apply the filter.
+			// Also if CKEDITOR.config.allowDangerousProtectedSource is enabled
+			// display warning inside the console
+			if ( editor.config.allowDangerousProtectedSource === true ) {
+				console.warn( 'Flag CKEDITOR.config.allowDangerousProtectedSource is set to true. This can potentially allow execution of malicious scripts inside the editor. Use with caution.' );
```

Comment:
Good idea, I wonder if it should be described in docs errors section (https://ckeditor.com/docs/ckeditor4/latest/guide/dev_errors.html) 🤔

Reply:
Where should I add docs for this?

Reply:
Added [pull request](https://github.com/ckeditor/ckeditor-docs/pull/251) to ckeditor-docs. Also replaced `console.warn` with `CKEDITOR.warn`.